src/validator/validator.ts

```diff
@@ -5,7 +5,7 @@ import type { Env, MiddlewareHandler, TypedResponse, ValidationTargets } from '.
 import type { BodyData } from '../utils/body'
 import { bufferToFormData } from '../utils/buffer'
 
-type ValidationTargetKeysWithBody = 'form' | 'json'
+type ValidationTargetKeysWithBody = 'form' | 'json' | 'formdot'
```

Comment:
At first glance, I can understand your needs, but adding the type `formdot` will make the type inference for RPC complicated. For example, `ValidationTargets` will have `formdot`. So, I prefer another way, though I don't imagine it now.

Reply:
You're right - I missed this as I was using `$url` and a `<form>` with preflattened `<inputs />`. Not something I'd usually do but I had a no-js on client requirement for this page.

I'm sure there's a type keyof magic to flatten nested objects, but it definitely is out of scope of this PR. I'll keep this in draft if you don't mind.

Reply:
@atoko 

Thank you. I want to close this. Okay?